File: backend/src/algoritmos/arrays/string_mas_largo_sin_repetir_caracter/longest.js
"use strict"

var longest_non_repeat = function(s){
	var start = 0,
	    maxlen = 0,
	    used_char = {};
	for (var i = 0; i < s.length; i++){
		var ch = s[i];
		if (ch in used_char && start <= used_char[ch]){
			start = used_char[ch] + 1
		} else {
			maxlen = Math.max(maxlen, i-start+1);
		}
		used_char[ch] = i
	}
	return maxlen
}

var a = "abcabcdefbb";
console.log(a);
console.log(longest_non_repeat(a)); // 6

File: backend/src/algoritmos/arrays/string_mas_largo_sin_repetir_caracter/longest.rb
#!/usr/bin/ruby

def longest_non_repeat(s)
	start, maxlen = [0, 0]
	used_char = {}
	s.split("").each_with_index do |char, i|
        if ( used_char.key?(char) && start <= used_char[char] )
        	start = used_char[char] + 1
        else
        	maxlen = [maxlen, i-start+1].max
        end
        used_char[char] = i
	end
	return maxlen
end


a = "abcabcdefbb"
p(a)
p(longest_non_repeat(a)) # 6